CHANGELOG.md

```diff
@@ -1,5 +1,30 @@
 # Changelog
 
+## Unreleased
+
+### Features
+
+- TTID/TTFD Improvements ([#2866](https://github.com/getsentry/sentry-dart/pull/2866))
+```dart
+// To report TTFD you need to use the SentryNavigatorObserver.
+ 
+// Before reporting long running work, you can get a reference to the currently displayed route.
+final sentryDisplay = Sentry.currentDisplay();
+
+// Then do some work like file I/O or networking....
+
+// Finally call `reportFullyDisplayed` on the sentryDisplay instance when the work operation is done.
+sentryDisplay?.reportFullyDisplayed()
+
+// Alternatively, you can wrap your widget in `SentryDisplayWidget`.
+
+SentryDisplayWidget(child: YourWidget())
+
+// Then call to finish TTFD when your long running work within your widget is done.
+
+SentryDisplayWidget.of(context).reportFullDisplay();
+```
```

Comment:
last comment for this: let's recommend using the widget and provide the manual api as an alternative